actionview/lib/action_view/template.rb

```diff
@@ -319,6 +319,13 @@ def locals_code
         # Only locals with valid variable names get set directly. Others will
         # still be available in local_assigns.
         locals = @locals - Module::RUBY_RESERVED_KEYWORDS
+        deprecated_locals = locals.grep(/\A@+/)
+        if deprecated_locals.any?
+          ActiveSupport::Deprecation.warn(<<~MSG)
+            Passing instance variables to `render` is deprecated.
+            In Rails 7.0, #{deprecated_locals.to_sentence} will be ignored.
```

Comment:
7.0 will be the next version, so maybe this should target 7.1?

Reply:
Hmm, yeah probably. :)
I cargo culted other deprecations which had 7.0 as well.
But those probably need to change as well.
I'll create a PR.

Reply:
Hey @ghiculescu. Yes I think would be an improvement.